### ISSUE TITLE
Example GSM Modem

### ISSUE THREAD

**Issue by spikeruk** [manuelquiero/SMS_Send_Receive#1]
Hi,
Do you have an example of the type of GSM modem this is for?
Would it work with one of those wavecom modems that can take 8 sim cards?

**Comment by manuelquiero** [comment]
Hi, yes it will work. The original code I based this project on was using
an 4-port (4-sim) GSM modem with Wavecom CPU

**Comment by spikeruk** [comment]
Thanks
Do you think if I plugged in 2 of these (sim800c usb) would it work?
That would presumably create 2 COM ports, would your code loop over those com ports and detect 2 GSM devices?

**Comment by manuelquiero** [comment]
Yes. I put a very specific deviceID that is unique to all GSM modems in
'where' clause of the linq in the Win32_PnpEntity in Encoder.cs file.
if you want to know more about the Win32_PnPEntity and other windows driver
related documentation, you can visit this link
https://learn.microsoft.com/en-us/windows/win32/cimwin32prov/win32-pnpentity

**Comment by spikeruk** [comment]
Yes
I've had a look at the code.
I think the issue will be if I plug in two of these sim800c USB devices that will show up as 2 COM ports, but your current code only expects one GSM modem COM port?.

I think it'll be fairly minor to change and i'll just need a separate worker thread for each COM device that is detected?

I have 2 sim800c usb devices currently ordered so will have a play when they arrive.
Your code is really helpful as I haven't done this sort of thing before.

**Comment by manuelquiero** [comment]
Yes. I've actually tested that code before. I plugged in two GSM modem
devices, one wavecom, and the other is a broadband stick, and yes, they
were both detected by the application.
This is because when I'm writing the code,  there was a possibility that
the company I'm working for at that time could buy another 4-port GSM modem
to expand the capacity of the application.

But if it didn't work, let me know.

I appreciate your positive inputs.
Thanks.

**Comment by spikeruk** [comment]
I'll test when both modems arrive but can't see how it will work as the code is as follows:

Connect button calls -> ConnectToModem() which then calls GetPortInfo()

However GetPortInfo() will only ever return the last port.
It has a foreach loop but the 'port' variable is overwritten each time.
Therefore I'm not sure how this version on Github could ever work with 2 modems?

**Comment by manuelquiero** [comment]
Ok, I'll check it out later.
